hazelcast-client/src/main/java/com/hazelcast/client/config/QueryCacheYamlConfigBuilderHelper.java

```diff
@@ -0,0 +1,97 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+import com.hazelcast.config.EntryListenerConfig;
+import com.hazelcast.config.InvalidConfigurationException;
+import com.hazelcast.config.MapIndexConfig;
+import com.hazelcast.config.PredicateConfig;
+import com.hazelcast.config.QueryCacheConfig;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+
+import static com.hazelcast.config.DomConfigHelper.childElements;
+import static com.hazelcast.config.DomConfigHelper.getBooleanValue;
+
+final class QueryCacheYamlConfigBuilderHelper extends AbstractQueryCacheConfigBuilderHelper {
+    QueryCacheYamlConfigBuilderHelper() {
+        super(true);
+    }
+
+    @Override
+    public void handleQueryCache(ClientConfig clientConfig, Node node) {
+        for (Node queryCacheNode : childElements(node)) {
+            handleQueryCacheNode(clientConfig, queryCacheNode);
+        }
+    }
+
+    @Override
+    protected String getCacheName(Node queryCacheNode) {
+        return queryCacheNode.getNodeName();
+    }
+
+    @Override
+    protected void handleEntryListeners(QueryCacheConfig queryCacheConfig, Node childNode) {
+        for (Node listenerNode : childElements(childNode)) {
+            NamedNodeMap attrs = listenerNode.getAttributes();
+            boolean incValue = getBooleanValue(getTextContent(attrs.getNamedItem("include-value")));
+            boolean local = getBooleanValue(getTextContent(attrs.getNamedItem("local")));
+            String listenerClass = getTextContent(attrs.getNamedItem("class-name"));
+            queryCacheConfig.addEntryListenerConfig(new EntryListenerConfig(listenerClass, local, incValue));
+        }
+    }
+
+    @Override
+    protected void queryCachePredicateHandler(Node childNode, QueryCacheConfig queryCacheConfig) {
+        NamedNodeMap predicateAttributes = childNode.getAttributes();
+        Node classNameNode = predicateAttributes.getNamedItem("class-name");
+        Node sqlNode = predicateAttributes.getNamedItem("sql");
+
+        if (classNameNode != null && sqlNode != null) {
+            throw new InvalidConfigurationException("Both class-name and sql is defined for the predicate of map "
+                    + childNode.getParentNode().getParentNode().getNodeName());
+        }
+
+        if (classNameNode == null && sqlNode == null) {
+            throw new InvalidConfigurationException("Either class-name and sql should be defined for the predicate of map "
+                    + childNode.getParentNode().getParentNode().getNodeName());
+        }
+
+        PredicateConfig predicateConfig = new PredicateConfig();
+        if (classNameNode != null) {
+            predicateConfig.setClassName(getTextContent(classNameNode));
+        } else if (sqlNode != null) {
+            predicateConfig.setSql(getTextContent(sqlNode));
```

Comment:
This seems not to be covered by any test. Could you check?

Reply:
Covered both in XML and YAML tests.